### DISCUSSION TITLE
How to give our own labels in the serviceMonitorSelector field of the prometheus deployment

### DISCUSSION THREAD

**Discussion by ar-sumesh** [prometheus-operator/kube-prometheus#1384] [Q&A]
i tried giving my labels like **app: myapp** in the serviceMonitorSelector field but prometheus is not able to do discovery of  targets , also not able to perform service discovery.
As i am not able to see the target and service discovery in the prometheus web page.

i done the modification in the `prometheus.libsonnet` , then complied them to get yaml files.

Also tell me how i can give these labels as `commonLabels`, so other components can also include this labels in  their deployment.

**Comment by paulfantom** [comment]
There is no need to modify `prometheus.libsonnet` as everything can be done from `main.jsonnet` file.

To add labels in selector (bear in mind that ALL objects selected need to have ALL labels from `serviceMonitorSelector` to be picked up by prometheus server):

```
{
  prometheus+: {
    prometheus+: {
      spec+: {
        serviceMonitorSelector+: {
          app: "myapp",
        }
     }
  }
}
```

To add labels to `commonLabels`:
```
{
  values+:: {
    prometheus+: {
      commonLabels+: {
        label: "value",
      },
    },
  },
}
